From 82f410643cb9bdb18be919c06e9a7cba5f2a3783 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 18 Mar 2024 15:00:55 +0000
Subject: [PATCH 0972/1002] ARM: dts: bcm2712: Move /soc/sound to bcm2712-rpi

The /soc/sound node is not a feature of the bcm2712 hardware, rather a
Raspberry Pi convention of where to declare soundcards. Move it to
bcm2712-rpi.dtsi accordingly, at the same time marking it as disabled
so that runtime overlay application causes the soundcard device to be
created.

See: https://forums.raspberrypi.com/viewtopic.php?t=367530

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi | 4 ++++
 arch/arm/boot/dts/broadcom/bcm2712.dtsi     | 3 ---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi b/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi
index 4c018cde845f..1f5d2bb29bfb 100644
--- a/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -158,6 +158,10 @@ gpiomem@7d510700 {
 		reg = <0x7d510700 0x20>;
 		chardev-name = "gpiomem4";
 	};
+
+	sound: sound {
+		status = "disabled";
+	};
 };
 
 i2c0: &rp1_i2c0 { };
diff --git a/arch/arm/boot/dts/broadcom/bcm2712.dtsi b/arch/arm/boot/dts/broadcom/bcm2712.dtsi
index 049cdfadbbbb..c4c9798e1d53 100644
--- a/arch/arm/boot/dts/broadcom/bcm2712.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2712.dtsi
@@ -720,9 +720,6 @@ hdmi1: hdmi@7ef05700 {
 			dma-names = "audio-rx";
 			status = "disabled";
 		};
-
-		sound: sound {
-		};
 	};
 
 	arm-pmu {
-- 
2.44.0

